### PR TITLE
Tools: wait for vehicle to disarm after handling throttle failsafe

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by magicrub** [comment]
Lgtm.

File: Tools/autotest/arducopter.py [hunk 1]
```diff
@@ -307,8 +307,13 @@ def fly_throttle_failsafe(mavproxy, mav, side=60, timeout=180):
             # switch back to stabilize
```
Comment (OXINARF) [inline Tools/autotest/arducopter.py]: Interesting comment. I can't figure out why we go to Land mode first - we do it if failsafe fails too.

**Comment by OXINARF** [comment]
@peterbarker Found another one that can fail: https://travis-ci.org/ArduPilot/ardupilot/jobs/259626416#L4606-L4617 This is in the *Test Horizontal Fence* test, can you do the same for it please?

**Comment by peterbarker** [comment]
On Tue, 1 Aug 2017, Francisco Ferreira wrote:

> Interesting comment. I can't figure out why we go to Land mode first - we do
> it if failsafe fails too.

Yeah, I don't know either.  I just tend to assume that the person knew 
what they were doing but hated comments.

**Comment by peterbarker** [comment]
On Tue, 1 Aug 2017, Francisco Ferreira wrote:

> @peterbarker Found another one that can fail:
> https://travis-ci.org/ArduPilot/ardupilot/jobs/259626416#L4606-L4617 This is
> in the Test Horizontal Fence test, can you do the same for it please?

Yeah, you made that sound easy :-)

Turns out we can't re-arm after having a battery failsafe event (which we 
test just before doing the horizontal-fence test) - at least without 
disabling arming checks.  So I did that.

I think we should reset the battery failsafe when we disarm.  The 
prearm/arm checks should ensure you don't take off with an exhausted 
battery.

Patches pushed up (@OXINARF)

**Comment by OXINARF** [comment]
@peterbarker Ups, sorry for the trouble 😃 I've done two changes (and squash them into your commits):

- in the first commit, change the *Reached home OK* message to be immediately after disarming motors (makes more sense and is coherent with the second commit)
- in the second commit, I added the re-enable of arming checks when we fail to re-arm; this is because other tests will continue and we want the checks enabled

Commits I plan to merge: https://github.com/OXINARF/ardupilot/compare/e0879c6...abfd52a83f45
Let me know if they look OK to you.

**Comment by peterbarker** [comment]
On Thu, 3 Aug 2017, Francisco Ferreira wrote:

>  *  in the first commit, change the Reached home OK message to be
>     immediately after disarming motors (makes more sense and is coherent
>     with the second commit)

Yeah, I was in two minds about this.  I actually decided reaching home was 
a separate thing to disarming.  But whatever :-)

>  *  in the second commit, I added the re-enable of arming checks when we
>     fail to re-arm; this is because other tests will continue and we want
>     the checks enabled

Gah!  Thanks!  It even ran through my head I'd have to handle both cases!

> 
> Commits I plan to merge: OXINARF/ardupilot@e0879c6...abfd52a
> Let me know if they look OK to you.

All good!

**Comment by OXINARF** [comment]
Merged my branch, thanks!

**Pull request closed without merging**